Makefile

```diff
@@ -0,0 +1,38 @@
+goimports := golang.org/x/tools/cmd/goimports@v0.1.12
+golangci_lint := github.com/golangci/golangci-lint/cmd/golangci-lint@v1.49.0
+
+.PHONY: test
+test:
+	@go test -v ./...
+
+golangci_lint_path := $(shell go env GOPATH)/bin/golangci-lint
+
+$(golangci_lint_path):
+	@go install $(golangci_lint)
+
+.PHONY: lint
+lint: $(golangci_lint_path)
+	@CGO_ENABLED=0 $(golangci_lint_path) run --timeout 5m
+
+.PHONY: format
+format:
+	@find . -type f -name '*.go' | xargs gofmt -s -w
+	@for f in `find . -name '*.go'`; do \
+	    awk '/^import \($$/,/^\)$$/{if($$0=="")next}{print}' $$f > /tmp/fmt; \
+	    mv /tmp/fmt $$f; \
+	done
+	@go run $(goimports) -w -local github.com/http-wasm/http-wasm-host-go `find . -name '*.go'`
+
+.PHONY: check
+check:
+	@$(MAKE) lint
```

Comment:
Curious about this syntax, what about 

```
check: lint format
    @go mod tidy
```

Reply:
conventionally, I've noticed folks put the prereqs as files not phonies. I'm not sure if phonies work on all make or if it is worth breaking convention over.